### ISSUE TITLE
[java] PMDException: Error while processing a source file

### ISSUE THREAD

**Issue by maisonobe** [pmd/pmd#4499]
**Affects PMD Version:** 

6.53.0 (it is the one referenced by the latest maven-pmd-plugin which is 3.20.0)

**Description:**

When running the tool on the Hipparchus library (file [FieldDerivativeStructure.java](https://github.com/Hipparchus-Math/hipparchus/blob/update-build-dependencies/hipparchus-core/src/main/java/org/hipparchus/analysis/differentiation/FieldDerivativeStructure.java)), an exception is triggered.

**Exception Stacktrace:**

```
<error filename="/home/luc/sources/eclipse/hipparchus/hipparchus-core/src/main/java/org/hipparchus/analysis/differentiation/FieldDerivativeStructure.java" msg="PMDException: Error while processing /home/luc/sources/eclipse/hipparchus/hipparchus-core/src/main/java/org/hipparchus/analysis/differentiation/FieldDerivativeStructure.java">
<![CDATA[net.sourceforge.pmd.PMDException: Error while processing /home/luc/sources/eclipse/hipparchus/hipparchus-core/src/main/java/org/hipparchus/analysis/differentiation/FieldDerivativeStructure.java
	at net.sourceforge.pmd.SourceCodeProcessor.processSourceCodeWithoutCache(SourceCodeProcessor.java:128)
	at net.sourceforge.pmd.SourceCodeProcessor.processSourceCode(SourceCodeProcessor.java:100)
	at net.sourceforge.pmd.SourceCodeProcessor.processSourceCode(SourceCodeProcessor.java:62)
	at net.sourceforge.pmd.processor.PmdRunnable.call(PmdRunnable.java:89)
	at net.sourceforge.pmd.processor.PmdRunnable.call(PmdRunnable.java:30)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalStateException: Object is not an array type!
	at net.sourceforge.pmd.lang.java.typeresolution.typedefinition.JavaTypeDefinitionSimple.getComponentType(JavaTypeDefinitionSimple.java:243)
	at net.sourceforge.pmd.lang.java.typeresolution.MethodType.getVarargComponentType(MethodType.java:63)
	at net.sourceforge.pmd.lang.java.typeresolution.MethodType.getArgTypeIncludingVararg(MethodType.java:74)
	at net.sourceforge.pmd.lang.java.typeresolution.MethodTypeResolution.checkSubtypeability(MethodTypeResolution.java:111)
	at net.sourceforge.pmd.lang.java.typeresolution.MethodTypeResolution.selectMostSpecificMethod(MethodTypeResolution.java:418)
	at net.sourceforge.pmd.lang.java.typeresolution.MethodTypeResolution.getBestMethodReturnType(MethodTypeResolution.java:388)
	at net.sourceforge.pmd.lang.java.typeresolution.ClassTypeResolver.visit(ClassTypeResolver.java:1019)
	at net.sourceforge.pmd.lang.java.ast.ASTPrimaryExpression.jjtAccept(ASTPrimaryExpression.java:25)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:62)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:13)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:298)
	at net.sourceforge.pmd.lang.java.typeresolution.ClassTypeResolver.visit(ClassTypeResolver.java:816)
	at net.sourceforge.pmd.lang.java.ast.ASTExpression.jjtAccept(ASTExpression.java:37)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:62)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:13)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:538)
	at net.sourceforge.pmd.lang.java.ast.ASTReturnStatement.jjtAccept(ASTReturnStatement.java:25)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:62)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:13)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:448)
	at net.sourceforge.pmd.lang.java.ast.ASTStatement.jjtAccept(ASTStatement.java:25)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:62)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:13)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:463)
	at net.sourceforge.pmd.lang.java.ast.ASTBlockStatement.jjtAccept(ASTBlockStatement.java:25)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:62)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:13)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:458)
	at net.sourceforge.pmd.lang.java.ast.ASTBlock.jjtAccept(ASTBlock.java:27)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:62)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:13)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:238)
	at net.sourceforge.pmd.lang.java.ast.ASTMethodDeclaration.jjtAccept(ASTMethodDeclaration.java:37)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:62)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:13)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:58)
	at net.sourceforge.pmd.lang.java.ast.ASTClassOrInterfaceBodyDeclaration.jjtAccept(ASTClassOrInterfaceBodyDeclaration.java:44)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:62)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:13)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:53)
	at net.sourceforge.pmd.lang.java.ast.ASTClassOrInterfaceBody.jjtAccept(ASTClassOrInterfaceBody.java:35)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:62)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:13)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:23)
	at net.sourceforge.pmd.lang.java.ast.ASTClassOrInterfaceDeclaration.jjtAccept(ASTClassOrInterfaceDeclaration.java:55)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:62)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:13)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:208)
	at net.sourceforge.pmd.lang.java.typeresolution.ClassTypeResolver.visit(ClassTypeResolver.java:258)
	at net.sourceforge.pmd.lang.java.ast.ASTTypeDeclaration.jjtAccept(ASTTypeDeclaration.java:39)
	at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.childrenAccept(AbstractJavaNode.java:62)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:13)
	at net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter.visit(JavaParserVisitorAdapter.java:183)
	at net.sourceforge.pmd.lang.java.typeresolution.ClassTypeResolver.visit(ClassTypeResolver.java:229)
	at net.sourceforge.pmd.lang.java.ast.ASTCompilationUnit.jjtAccept(ASTCompilationUnit.java:44)
	at net.sourceforge.pmd.lang.java.typeresolution.TypeResolutionFacade.initializeWith(TypeResolutionFacade.java:21)
	at net.sourceforge.pmd.lang.java.AbstractJavaHandler$5.start(AbstractJavaHandler.java:121)
	at net.sourceforge.pmd.SourceCodeProcessor.usesTypeResolution(SourceCodeProcessor.java:178)
	at net.sourceforge.pmd.SourceCodeProcessor.processSource(SourceCodeProcessor.java:205)
	at net.sourceforge.pmd.SourceCodeProcessor.processSourceCodeWithoutCache(SourceCodeProcessor.java:118)
	... 10 more
]]>
</error>
```

**Code Sample demonstrating the issue:**

 [FieldDerivativeStructure.java](https://github.com/Hipparchus-Math/hipparchus/blob/update-build-dependencies/hipparchus-core/src/main/java/org/hipparchus/analysis/differentiation/FieldDerivativeStructure.java)

**Steps to reproduce:**

Beware: the following steps are only applicable to the update-build-dependencies branch, the other branches
use older versions of the tools.

1.  mvn clean
2.  mvn -Prelease site site:stage

The error was triggered on a Linux machine running Debian testing with openjdk 17.

**Comment by jsotuyod** [comment]
This is most probably fixed in PMD 7.0.0-rc1, as we rewrote all the typeresolution code.

You can check [here](https://github.com/pmd/pmd/discussions/4478#discussioncomment-5641275) how to set up a compatible version of the PMD Maven plugin.

**Comment by maisonobe** [comment]
The exceptions from my initial post are corrected with PMD 7 and the snapshot maven plugin.

However, I get a different set of exceptions with PMD 7.0.0:

```
[WARNING] Exception applying rule ImplicitSwitchFallThrough on file file:///home/luc/sources/eclipse/hipparchus/hipparchus-ode/src/main/java/org/hipparchus/ode/events/FilterType.java, continuing with next rule
org.apache.commons.lang3.exception.ContextedRuntimeException: java.lang.NullPointerException
Exception Context:
	[1:Rule applied on node=!debug only! [SwitchStatement:152:17]switch (previous) {
                    case UNINITIALIZED :
                        // we are initializing the first point
               (truncated)]
---------------------------------
    at net.sourceforge.pmd.util.AssertionUtil.contexted (AssertionUtil.java:232)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.applyOnIndex (RuleApplicator.java:77)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.apply (RuleApplicator.java:55)
    at net.sourceforge.pmd.RuleSets.apply (RuleSets.java:158)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.processSource (PmdRunnable.java:140)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.run (PmdRunnable.java:80)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:539)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    at java.lang.Thread.run (Thread.java:833)
Caused by: java.lang.NullPointerException
    at java.util.Objects.requireNonNull (Objects.java:208)
    at net.sourceforge.pmd.lang.java.rule.internal.DataflowPass$DataflowResult.switchBranchFallsThrough (DataflowPass.java:263)
    at net.sourceforge.pmd.lang.java.rule.errorprone.ImplicitSwitchFallThroughRule.visit (ImplicitSwitchFallThroughRule.java:39)
    at net.sourceforge.pmd.lang.java.ast.ASTSwitchStatement.acceptVisitor (ASTSwitchStatement.java:20)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.apply (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.rule.AbstractDelegateRule.apply (AbstractDelegateRule.java:238)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.applyOnIndex (RuleApplicator.java:75)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.apply (RuleApplicator.java:55)
    at net.sourceforge.pmd.RuleSets.apply (RuleSets.java:158)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.processSource (PmdRunnable.java:140)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.run (PmdRunnable.java:80)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:539)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    at java.lang.Thread.run (Thread.java:833)
[WARNING] Exception occurred on node !debug only! [SwitchStatement:152:17]switch (previous) {
                    case UNINITIALIZED :
                        // we are initializing the first point
               (truncated)
[WARNING] Exception applying rule ImplicitSwitchFallThrough on file file:///home/luc/sources/eclipse/hipparchus/hipparchus-ode/src/main/java/org/hipparchus/ode/events/FilterType.java, continuing with next rule
org.apache.commons.lang3.exception.ContextedRuntimeException: java.lang.NullPointerException
Exception Context:
	[1:Rule applied on node=!debug only! [SwitchStatement:275:17]switch (previous) {
                    case UNINITIALIZED :
                        // we are initializing the first point
               (truncated)]
---------------------------------
    at net.sourceforge.pmd.util.AssertionUtil.contexted (AssertionUtil.java:232)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.applyOnIndex (RuleApplicator.java:77)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.apply (RuleApplicator.java:55)
    at net.sourceforge.pmd.RuleSets.apply (RuleSets.java:158)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.processSource (PmdRunnable.java:140)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.run (PmdRunnable.java:80)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:539)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    at java.lang.Thread.run (Thread.java:833)
Caused by: java.lang.NullPointerException
    at java.util.Objects.requireNonNull (Objects.java:208)
    at net.sourceforge.pmd.lang.java.rule.internal.DataflowPass$DataflowResult.switchBranchFallsThrough (DataflowPass.java:263)
    at net.sourceforge.pmd.lang.java.rule.errorprone.ImplicitSwitchFallThroughRule.visit (ImplicitSwitchFallThroughRule.java:39)
    at net.sourceforge.pmd.lang.java.ast.ASTSwitchStatement.acceptVisitor (ASTSwitchStatement.java:20)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.apply (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.rule.AbstractDelegateRule.apply (AbstractDelegateRule.java:238)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.applyOnIndex (RuleApplicator.java:75)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.apply (RuleApplicator.java:55)
    at net.sourceforge.pmd.RuleSets.apply (RuleSets.java:158)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.processSource (PmdRunnable.java:140)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.run (PmdRunnable.java:80)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:539)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    at java.lang.Thread.run (Thread.java:833)
[WARNING] Exception occurred on node !debug only! [SwitchStatement:275:17]switch (previous) {
                    case UNINITIALIZED :
                        // we are initializing the first point
               (truncated)
[WARNING] Exception applying rule ImplicitSwitchFallThrough on file file:///home/luc/sources/eclipse/hipparchus/hipparchus-ode/src/main/java/org/hipparchus/ode/events/FilterType.java, continuing with next rule
org.apache.commons.lang3.exception.ContextedRuntimeException: java.lang.NullPointerException
Exception Context:
	[1:Rule applied on node=!debug only! [SwitchStatement:97:17]switch (previous) {
                    case UNINITIALIZED :
                        // we are initializing the first point
               (truncated)]
---------------------------------
    at net.sourceforge.pmd.util.AssertionUtil.contexted (AssertionUtil.java:232)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.applyOnIndex (RuleApplicator.java:77)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.apply (RuleApplicator.java:55)
    at net.sourceforge.pmd.RuleSets.apply (RuleSets.java:158)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.processSource (PmdRunnable.java:140)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.run (PmdRunnable.java:80)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:539)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    at java.lang.Thread.run (Thread.java:833)
Caused by: java.lang.NullPointerException
    at java.util.Objects.requireNonNull (Objects.java:208)
    at net.sourceforge.pmd.lang.java.rule.internal.DataflowPass$DataflowResult.switchBranchFallsThrough (DataflowPass.java:263)
    at net.sourceforge.pmd.lang.java.rule.errorprone.ImplicitSwitchFallThroughRule.visit (ImplicitSwitchFallThroughRule.java:39)
    at net.sourceforge.pmd.lang.java.ast.ASTSwitchStatement.acceptVisitor (ASTSwitchStatement.java:20)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.apply (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.rule.AbstractDelegateRule.apply (AbstractDelegateRule.java:238)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.applyOnIndex (RuleApplicator.java:75)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.apply (RuleApplicator.java:55)
    at net.sourceforge.pmd.RuleSets.apply (RuleSets.java:158)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.processSource (PmdRunnable.java:140)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.run (PmdRunnable.java:80)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:539)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    at java.lang.Thread.run (Thread.java:833)
[WARNING] Exception occurred on node !debug only! [SwitchStatement:97:17]switch (previous) {
                    case UNINITIALIZED :
                        // we are initializing the first point
               (truncated)
[WARNING] Exception applying rule ImplicitSwitchFallThrough on file file:///home/luc/sources/eclipse/hipparchus/hipparchus-ode/src/main/java/org/hipparchus/ode/events/FilterType.java, continuing with next rule
org.apache.commons.lang3.exception.ContextedRuntimeException: java.lang.NullPointerException
Exception Context:
	[1:Rule applied on node=!debug only! [SwitchStatement:330:17]switch (previous) {
                    case UNINITIALIZED :
                        // we are initializing the first point
               (truncated)]
---------------------------------
    at net.sourceforge.pmd.util.AssertionUtil.contexted (AssertionUtil.java:232)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.applyOnIndex (RuleApplicator.java:77)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.apply (RuleApplicator.java:55)
    at net.sourceforge.pmd.RuleSets.apply (RuleSets.java:158)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.processSource (PmdRunnable.java:140)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.run (PmdRunnable.java:80)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:539)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    at java.lang.Thread.run (Thread.java:833)
Caused by: java.lang.NullPointerException
    at java.util.Objects.requireNonNull (Objects.java:208)
    at net.sourceforge.pmd.lang.java.rule.internal.DataflowPass$DataflowResult.switchBranchFallsThrough (DataflowPass.java:263)
    at net.sourceforge.pmd.lang.java.rule.errorprone.ImplicitSwitchFallThroughRule.visit (ImplicitSwitchFallThroughRule.java:39)
    at net.sourceforge.pmd.lang.java.ast.ASTSwitchStatement.acceptVisitor (ASTSwitchStatement.java:20)
    at net.sourceforge.pmd.lang.java.ast.AbstractJavaNode.acceptVisitor (AbstractJavaNode.java:38)
    at net.sourceforge.pmd.lang.java.rule.AbstractJavaRule.apply (AbstractJavaRule.java:24)
    at net.sourceforge.pmd.lang.rule.AbstractDelegateRule.apply (AbstractDelegateRule.java:238)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.applyOnIndex (RuleApplicator.java:75)
    at net.sourceforge.pmd.lang.rule.internal.RuleApplicator.apply (RuleApplicator.java:55)
    at net.sourceforge.pmd.RuleSets.apply (RuleSets.java:158)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.processSource (PmdRunnable.java:140)
    at net.sourceforge.pmd.lang.impl.PmdRunnable.run (PmdRunnable.java:80)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:539)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    at java.lang.Thread.run (Thread.java:833)
[WARNING] Exception occurred on node !debug only! [SwitchStatement:330:17]switch (previous) {
                    case UNINITIALIZED :
                        // we are initializing the first point
               (truncated)
[WARNING] There are 4 PMD processing errors:
[WARNING] /home/luc/sources/eclipse/hipparchus/hipparchus-ode/src/main/java/org/hipparchus/ode/events/FilterType.java: ContextedRuntimeException: java.lang.NullPointerException
Exception Context:
	[1:Rule applied on node=!debug only! [SwitchStatement:152:17]switch (previous) {
                    case UNINITIALIZED :
                        // we are initializing the first point
               (truncated)]
---------------------------------
/home/luc/sources/eclipse/hipparchus/hipparchus-ode/src/main/java/org/hipparchus/ode/events/FilterType.java: ContextedRuntimeException: java.lang.NullPointerException
Exception Context:
	[1:Rule applied on node=!debug only! [SwitchStatement:275:17]switch (previous) {
                    case UNINITIALIZED :
                        // we are initializing the first point
               (truncated)]
---------------------------------
/home/luc/sources/eclipse/hipparchus/hipparchus-ode/src/main/java/org/hipparchus/ode/events/FilterType.java: ContextedRuntimeException: java.lang.NullPointerException
Exception Context:
	[1:Rule applied on node=!debug only! [SwitchStatement:97:17]switch (previous) {
                    case UNINITIALIZED :
                        // we are initializing the first point
               (truncated)]
---------------------------------
/home/luc/sources/eclipse/hipparchus/hipparchus-ode/src/main/java/org/hipparchus/ode/events/FilterType.java: ContextedRuntimeException: java.lang.NullPointerException
Exception Context:
	[1:Rule applied on node=!debug only! [SwitchStatement:330:17]switch (previous) {
                    case UNINITIALIZED :
                        // we are initializing the first point
               (truncated)]
---------------------------------
[INFO] PMD version: 7.0.0-SNAPSHOT
```

Note that I did not fix the warnings from my 6.x.x version of `mpd-ruleset.xml` before running the check, but I doubt these warning are related to the exceptions.
